docs/DafnyRef/Types.md

```diff
@@ -3353,7 +3353,8 @@ befuddlement is to write `array<T>` instead of `T` after `new`.
 For example, consider the following:
 <!-- %check-resolve Types.17.expect -->
 ```dafny
-class A {
+type T(0)
+method m(n: nat) {
   var a := new array<T>;
   var b := new array<T>[n];
   var c := new array<T>(n);  // resolution error
```

Comment:
I don't know the context of these lines, but note that this `b` is an array of arrays (that is, the assignment makes the type of `b` an `array<array<T>>`).
Also, `a` is assigned a object of type `array<T>`, but nothing is said about the length of that array. It would probably be better for Dafny to disallow this.

Reply:
The context was a discussion of just those points (of possible user confusion). 

The reason they show up in this PR is that I had incorrectly embodied the declaration statements in a class (and thus got the wrong error messages) and working out the error messages about initializers identified my error.